### ISSUE TITLE
No usable method for LoadImageFromMemory

### ISSUE THREAD

**Issue by JupiterRider** [ChrisDill/Raylib-cs#108]
Hey guys,

there is no C# binding for `LoadImageFromMemory`. 

Something like this should fit:
```
    [DllImport("raylib", CallingConvention = CallingConvention.Cdecl)]
    public static extern Image LoadImageFromMemory(string fileType, byte[] fileData, int dataSize);
```

Btw: Do we really need the partial class Raylib.Utils.cs? The code above works as well. 

Thanks!

**Comment by JupiterRider** [comment]
Oh, I see, there are much more. Seems like all Load***FromMemory bindings are missed.

**Comment by ChrisDill** [comment]
There is a unsafe version of `LoadImageFromMemory` you can use.
I will consider adding utils for other LoadXFromMemory functions.

**Comment by JupiterRider** [comment]
@ChrisDill 
Is there any downside of just putting the save code into Raylib.cs? Is unsafe interop faster?

**Comment by ChrisDill** [comment]
We decided to split the safe utils from the unsafe bindings to make it more flexible and easier to debug. Performance depends on your use case.

**Comment by JupiterRider** [comment]
@ChrisDill 

Would something like this fit your needs:
```csharp
public static extern Image LoadImageFromMemory(string fileType, [MarshalAs(UnmanagedType.LPArray)] byte[] fileData, int dataSize);
```

**Comment by ChrisDill** [comment]
@JupiterRider The issue is the `Image` type uses unmanaged memory which must be handled manually. The safe utils were also split from the bindings to make it clearer what is being marshaled behaviour explicit.

**Comment by ChrisDill** [comment]
@JupiterRider Raylib-cs 4.5.0.1 now includes utils for the existing LoadXFromMemory bindings.

**Comment by JupiterRider** [comment]
@ChrisDill Tanks.

Oh I see, u are creating a new C array with `malloc()` and copy the values over. 
But I think you forgot to `free()` the memory again. 

Here is an example of raylib. They free the file data too:
https://github.com/raysan5/raylib/blob/master/src/rtextures.c#L255

What do u think about using the fixed statement instead?
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/fixed

**Comment by ChrisDill** [comment]
Thanks. Updated to use fixed statements. I assumed it was stored in the result and not copied internally.

**Comment by JupiterRider** [comment]
@ChrisDill Thanks!

I have one final question:
Why do we use `Marshal.StringToCoTaskMemUTF8(String)` to marshal strings?

What is the difference between that and this for example?
```csharp
public static extern void SetTitle([MarshalAs(UnmanagedType.LPUTF8Str)] string title)
```

https://learn.microsoft.com/en-us/dotnet/framework/interop/default-marshalling-for-strings

**Comment by ChrisDill** [comment]
Fix for LoadXFromMemory utils now released in 4.2.0.2.

@JupiterRider Both string approaches can work. Found it more flexible to have both a more direct version and a version that adds marshalling code to make it easier to use.
